src/bootstrap/native.rs

```diff
@@ -307,38 +307,39 @@ fn download_component(builder: &Builder<'_>, base: &str, url: &str, dest_path: &
 fn download_with_retries(builder: &Builder<'_>, tempfile: &str, url: &str) {
     println!("downloading {}", url);
 
-    // FIXME: check if curl is installed instead of skipping straight to powershell
-    if builder.build.build.contains("windows-msvc") {
-        for _ in 0..3 {
-            if builder.try_run(Command::new("PowerShell.exe").args(&[
-                "/nologo",
-                "-Command",
-                "[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12;",
-                &format!(
-                    "(New-Object System.Net.WebClient).DownloadFile('{}', '{}')",
-                    url, tempfile
-                ),
-            ])) {
-                return;
+    // Try curl, if that fail and we are on windows
+    // fallback to PowerShell
+    if !builder.try_run(Command::new("curl").args(&[
```

Comment:
How does this behave on Linux if curl fails? I think currently it will fail the download, but still continue the build, which is not correct.

Reply:
Oh good point! It's indeed what the original code is doing. Will fix!

Reply:
Added the `check_run` who behave like `try_run` but doesn't terminate